server/protocols/protocols-smtp-dkim/src/main/java/org/apache/james/smtpserver/DKIMHook.java

```diff
@@ -92,6 +104,35 @@ static DKIMCheckNeeded onlyForSenderDomain(Domain domain) {
                 .orElse(false);
         }
 
+        static DKIMCheckNeeded onlyForHeaderFromDomain(Domain domain) {
+            return mail -> {
+                try {
+                    return StreamUtils.ofNullable(mail.getMessage().getFrom())
+                        .distinct()
+                        .flatMap(DKIMCheckNeeded::parseMailAddress)
+                        .findFirst()
+                        .map(MailAddress::getDomain)
+                        .map(domain::equals)
+                        .orElse(false);
+                } catch (MessagingException me) {
+                    LOGGER.info("Unable to parse the \"FROM\" header; ignoring.");
```

Comment:
```suggestion
                    LOGGER.info("Unable to parse the \"FROM\" header {}; ignoring.", message.getHeader("From"));
```

Reply:
Yep but getting the actual failing value is worthwhile: being the one reading prod logs I can tell you I want this. Note also I am not asking for parsed form but row form which is unlikely to fail...